storage/rocksdb/ha_rocksdb.cc

```diff
@@ -6003,37 +6082,72 @@ int ha_rocksdb::read_key_exact(const Rdb_key_def &kd,
     lookup_tuple may be a prefix of the index.
   */
   rocksdb_smart_seek(kd.m_is_reverse_cf, iter, key_slice);
-  if (!iter->Valid() || !kd.value_matches_prefix(iter->key(), key_slice)) {
+
+  for (;;) {
```

Comment:
Nice, I took the idea and came out with this instead, which I think is even cleaner

```  
while (iter->Valid() && kd.value_matches_prefix(iter->key(), key_slice)) {
   
    if (kd.has_ttl() &&
        should_hide_ttl_rec(iter->value(), tx->m_snapshot_timestamp)) {
      rocksdb_smart_next(kd.m_is_reverse_cf, iter);
      continue;
    }

    return HA_EXIT_SUCCESS;
 }

return HA_ERR_KEY_NOT_FOUND;
```